imap_processing/tests/mag/test_mag_l1a.py

```diff
@@ -47,7 +305,364 @@ def test_compare_validation_data():
         assert l1_magi.vectors[index][3] == validation_data["rng_sec"][index]
 
 
-def test_process_vector_data():
+def test_compressed_vector_data(expected_vectors):
+    # Values from test packet
+    primary_expected = expected_vectors[0]
+    secondary_expected = expected_vectors[1]
+
+    # Compression headers - indicating a 16 bit width and no range section
+    headers = "01000000"
+
+    # bit width of 16, generated from primary_expected and secondary_expected using
+    # encoding code from MAG team
+    # Validated with decoding code from mag team
+    primary_compressed = (
+        "000000100000010000001000000100000001000000100000110101110010"
+        "011100101011010111001001110010101101011100100110000000011100"
+        "011100100111001010110101100001011000000001101011100100111001"
+        "010111000111001001110010101101011100100110000000011010111001"
+        "001110010101101011100100111001010111000110000101100000000110"
+        "101110010011100101011010111001001110010101110001110010011000"
+        "00000110101110010011100101011"
+    )
+
+    secondary_compressed = (
+        "0000001000000011000010000000111100010000000111111110001110"
+        "0100111001010110101100001011000000001101011100100111001010"
+        "1110001110010011100101011010111001001100000000110101110010"
+        "0111001010110101110010011100101011100011000010110000000011"
+        "0101110010011100101011010111001001100000000111000111001001"
+        "1100101011010111001001110010101101011000010110000000011100"
+        "011100100111001010110101110010011100101011"
+    )
+
+    padding = "00000"  # Pad to byte boundary
+    input_data = np.array(
+        [int(i) for i in headers + primary_compressed + secondary_compressed + padding],
+        dtype=np.uint8,
+    )
+
+    # Will be the input data format
+    input_data = np.packbits(input_data)
+
+    (primary, secondary) = MagL1a.process_compressed_vectors(input_data, 16, 16)
+
+    assert np.array_equal(primary[0], primary_expected[0])
+    assert np.array_equal(secondary[0], secondary_expected[0])
+
+    assert primary.shape[0] == 16
+    assert secondary.shape[0] == 16
+
+    assert primary.shape[1] == 4
+    assert secondary.shape[1] == 4
+
+    assert np.array_equal(primary, primary_expected)
+    assert np.array_equal(secondary, secondary_expected)
+
+    # range data has 2 bits per vector, primary and then secondary in sequence.
+    # It excludes the first vector, making the length (primary_count - 1) * 2
+    range_primary = "000000000000000101101011111111"
+
+    expected_range_primary = [3, 0, 0, 0, 0, 0, 0, 0, 1, 1, 2, 2, 3, 3, 3, 3]
```

Comment:
```suggestion
    first_primary_vector_range = 3  # This is not included in the range data section
    expected_range_primary = [first_primary_vector_range, 0, 0, 0, 0, 0, 0, 0, 1, 1, 2, 2, 3, 3, 3, 3]
```
Just for readability